src/lsm/compaction.zig

```diff
@@ -463,6 +476,82 @@ pub fn CompactionType(
             }
         }
 
+        // TODO: We need to reimplement the .secondary_index optimization for sort here.
+        /// Copies values to `target` from our immutable table input, taking the last
+        /// matching value to resolve duplicates, and updating the immutable table slice.
+        fn fill_immutable_values(compaction: *Compaction, target: []Value) usize {
+            var source = compaction.context.table_info_a.immutable;
+            assert(source.len > 0);
+
+            if (constants.verify) {
+                // The input may have duplicate keys (last one wins), but keys must be
+                // non-decreasing.
+                // A source length of 1 is always non-decreasing.
+                if (source.len > 1) {
```

Comment:
I think this `source.len > 1` condition is unnecessary and the for-assert within can succeed regardless.

Reply:
> and the for-assert within can succeed

... no matter which way you slice it :laughing: 

But yes.